server-spi-private/src/main/java/org/keycloak/services/resource/AccountResourceProviderFactory.java

```diff
@@ -0,0 +1,33 @@
+package org.keycloak.services.resource;
+
+import jakarta.ws.rs.InternalServerErrorException;
+import jakarta.ws.rs.NotFoundException;
+import java.io.IOException;
+import org.keycloak.models.ClientModel;
+import org.keycloak.models.Constants;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.provider.ProviderFactory;
+import org.keycloak.theme.Theme;
+
+/**
+ * <p>A factory that creates {@link AccountResourceProvider} instances.
+ */
+public interface AccountResourceProviderFactory extends ProviderFactory<AccountResourceProvider> {
+
+  default Theme getTheme(KeycloakSession session) {
```

Comment:
Does it make more sense to start without these methods?

Looks like we can just move them for now to the default factory implementation as it is not being used outside the class hierarchy.

Reply:
I use them here https://github.com/keycloak/keycloak/blob/3a6c82ad3381c0250c247a0274c25e7a2e64ff26/services/src/main/java/org/keycloak/services/resources/account/AccountConsoleFactory.java#L31 and in the example implementation I did to bring back the v1 account console. I thought they would be useful helper methods for implementors. Do you think there is another place they could/should live?